packages/eslint-config-react-app/base.js

```diff
@@ -41,8 +50,10 @@ module.exports = {
   },
 
   rules: {
-    'react/jsx-uses-react': 'warn',
     'react/jsx-uses-vars': 'warn',
-    'react/react-in-jsx-scope': 'error',
+    ...(!hasJsxRuntime && {
+      'react/jsx-uses-react': 'warn',
```

Comment:
According to the [blog post](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports), it won't hurt if users still import React, however they don't _need_ to do it if the version of React they have supports the new transform.

So I think it's correct to flag `React` as an unused import in this scenario.

Reply:
It's also worth noting that this may mislead users who have the new transform installed, but aren't using Create React App and haven't configured Babel for the new transform.

If we release the ESLint config as a major, and add that to the release notes, I think we'll be fine.

Reply:
Note that opening a project to see a hundred lint warnings would be a bad experience. Especially if people don’t realize there is a codemod they can run. They will also be copy-pasting code from the web that has this import. How do we plan to explain this?

Reply:
I almost think it shouldn’t warn in the transitional period. We can tighten it up later as a separate breaking change.

Reply:
I had it that way - then flipped it, as I wasn't sure. We can swap that back of course.

This will be a major version, so we do have the opportunity to detail these changes (and why) in the release notes - but I understand not everyone reads those. We could also look to add a custom ESLint rule for this - but that seems like overkill.

Reply:
This has been changed as a part of #9751, @gaearon.

We can change this later, and I'll reach out to the `eslint-plugin-react` team about the messaging.